UnityAds/UnityFullscreenAdAdapter.m

```diff
@@ -0,0 +1,184 @@
+@dynamic delegate;
+@dynamic localExtras;
+@dynamic hasAdAvailable;
+
+- (NSString *)getAdNetworkId {
+    return (self.placementId != nil) ? self.placementId : @"";
+}
+
+- (NSString *)adType {
+    return nil;
+ }
+
+#pragma mark - MPFullscreenAdAdapter Override
+
+- (BOOL)enableAutomaticImpressionAndClickTracking {
+    return NO;
+}
+
+- (BOOL)isRewardExpected {
+    return NO;
+}
+
+- (BOOL)hasAdAvailable {
+    return _placementId != nil;
+}
+
+- (void)requestAdWithAdapterInfo:(NSDictionary *)info adMarkup:(NSString *)adMarkup
+{
+    NSString *gameId = [info objectForKey:kMPUnityRewardedVideoGameId];
+   
+    if (gameId == nil) {
+        NSError *error = [NSError errorWithAdAdapterErrorCode: MPAdAdapterErrorCodeInvalidAdapter
+                                                  description: @"Custom event class data did not contain gameId."
+                                                   suggestion: @"Update your MoPub custom event class data to contain a valid Unity Ads gameId."];
+        MPLogAdEvent([MPLogEvent adLoadFailedForAdapter:NSStringFromClass(self.class) error:error], [self getAdNetworkId]);
+        [self.delegate fullscreenAdAdapter:self didFailToLoadAdWithError:error];
+        return;
+    }
+
+    // Only need to cache game ID for SDK initialization
+    [UnityAdsAdapterConfiguration updateInitializationParameters:info];
+
+    NSString *placementId = [info objectForKey:kUnityAdsOptionPlacementIdKey];
+    if (placementId == nil) {
+        placementId = [info objectForKey:kUnityAdsOptionZoneIdKey];
+    }
+
+    if (placementId == nil) {
+        NSError *error = [NSError errorWithAdAdapterErrorCode: MPAdAdapterErrorCodeInvalidAdapter
+                                                  description: @"Custom event class data did not contain placementId."
+                                                   suggestion: @"Update your MoPub custom event class data to contain a valid Unity Ads placementId."];
+        MPLogAdEvent([MPLogEvent adLoadFailedForAdapter:NSStringFromClass(self.class) error:error], placementId);
+        [self.delegate fullscreenAdAdapter:self didFailToLoadAdWithError:error];
+        return;
+    }
+    
+    UADSLoadOptions *options = [UADSLoadOptions new];
+    if (adMarkup != nil) {
+        _objectId = [[NSUUID UUID] UUIDString];
+        [options setObjectId:_objectId];
+        [options setAdMarkup:adMarkup];
+    }
+    
+    MPLogAdEvent([MPLogEvent adLoadAttemptForAdapter:NSStringFromClass(self.class) dspCreativeId:nil dspName:nil], placementId);
+    [UnityAds load:placementId options:options loadDelegate:self];
+}
+
+- (void)presentAdFromViewController:(UIViewController *)viewController
+{
+    if (![self hasAdAvailable]) {
+        MPLogWarn(@"Unity Ads received call to show before successfully loading an ad");
+    }
+    
+    UADSShowOptions *options = [UADSShowOptions new];
+    if (_objectId != nil) {
+        [options setObjectId:_objectId];
```

Comment:
What will happen if objectId is nil?

Reply:
then it's not header bidding, should be working like it is now